From f1ad51b615599daebae19710c28e49f75d38c43c Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 24 Jun 2025 10:49:40 +0800
Subject: [PATCH 23/63] loonggpu: Remove the check against moving pinned BOs

The check is already in the generic code.

Following the reference implementation change.

Link: https://git.kernel.org/torvalds/c/f87c1f0b7b79
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_ttm.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/loonggpu/loonggpu_ttm.c b/loonggpu/loonggpu_ttm.c
index c8f9c37..3b601d8 100644
--- a/loonggpu/loonggpu_ttm.c
+++ b/loonggpu/loonggpu_ttm.c
@@ -634,10 +634,7 @@ static int loonggpu_bo_move(struct ttm_buffer_object *bo, bool evict,
                         return r;
         }
 
-	/* Can't move a pinned BO */
 	abo = ttm_to_loonggpu_bo(bo);
-	if (WARN_ON_ONCE(lg_bo_pin_count(abo) > 0))
-		return -EINVAL;
 	adev = loonggpu_ttm_adev(bo->bdev);
 
 	if (!old_mem || (old_mem->mem_type == TTM_PL_SYSTEM && bo->ttm == NULL)) {
-- 
2.52.0

